app-api/src/test/java/org/openmrs/eip/app/route/sender/DbSyncRouteTest.java

```diff
@@ -2,53 +2,62 @@
-	
-	@Override
-	public String getTestRouteFilename() {
-		return "db-sync-route";
-	}
-	
-	@Test
-	public void shouldCreateAndSaveTheSenderSyncMessageForTheEvent() {
-		final int count = TestUtils.getEntities(SenderSyncMessage.class).size();
-		DefaultExchange exchange = new DefaultExchange(camelContext);
-		final String table = "visit";
-		final String uuid = "some-uuid";
-		final String op = "c";
-		final String requestUuid = "some-request-uuid";
-		final boolean snapshot = true;
-		Event event = createEvent(table, null, uuid, op);
-		event.setSnapshot(snapshot);
-		event.setRequestUuid(requestUuid);
-		exchange.setProperty(EX_PROP_EVENT, event);
-		
-		producerTemplate.send(URI_DBSYNC, exchange);
-		
-		List<SenderSyncMessage> syncMsgs = TestUtils.getEntities(SenderSyncMessage.class);
-		assertEquals(count + 1, syncMsgs.size());
-		SenderSyncMessage msg = syncMsgs.get(syncMsgs.size() - 1);
-		assertEquals(table, msg.getTableName());
-		assertEquals(uuid, msg.getIdentifier());
-		assertEquals(op, msg.getOperation());
-		assertEquals(snapshot, msg.getSnapshot());
-		assertEquals(requestUuid, msg.getRequestUuid());
-		assertNotNull(msg.getDateCreated());
-		assertNotNull(msg.getMessageUuid());
-	}
-	
+
+    @Override
+    public String getTestRouteFilename() {
+        return "db-sync-route";
+    }
+
+    @Test
+    public void shouldCreateAndSaveTheSenderSyncMessageForTheEvent() {
+        final int count = TestUtils.getEntities(SenderSyncMessage.class).size();
+        DefaultExchange exchange = new DefaultExchange(camelContext);
+        final String table = "visit";
+        final String uuid = "some-uuid";
+        final String op = "c";
+        final String requestUuid = "some-request-uuid";
+        final boolean snapshot = true;
+        DebeziumEvent debeziumEvent = this.createDebeziumEvent(table, requestUuid, uuid, op);
+        debeziumEvent.setDateCreated(new Date());
+        debeziumEvent.getEvent().setSnapshot(snapshot);
+        debeziumEvent.getEvent().setRequestUuid(requestUuid);
+        exchange.setProperty(EX_PROP_EVENT, debeziumEvent.getEvent());
+        exchange.setProperty(EX_PROP_DBZM_EVENT, debeziumEvent);
+
+        producerTemplate.send(URI_DBSYNC, exchange);
+
+        List<SenderSyncMessage> syncMsgs = TestUtils.getEntities(SenderSyncMessage.class);
+        assertEquals(count + 1, syncMsgs.size());
+        SenderSyncMessage msg = syncMsgs.get(syncMsgs.size() - 1);
+        assertEquals(table, msg.getTableName());
+        assertEquals(uuid, msg.getIdentifier());
+        assertEquals(op, msg.getOperation());
+        assertEquals(snapshot, msg.getSnapshot());
+        assertEquals(requestUuid, msg.getRequestUuid());
+        assertNotNull(msg.getDateCreated());
+        assertNotNull(msg.getMessageUuid());
+        assertNull(msg.getDateSent());
+        assertNotNull(msg.getEventDate());
+        assertEquals(msg.getEventDate().getTime(), debeziumEvent.getDateCreated().getTime());
```

Comment:
Can't you just compare the dates without calling getTime()

Reply:
Without calling getTime method i had this result: 
java.lang.AssertionError: expected:<2022-10-07 19:34:08.758> but was:<Fri Oct 07 19:34:08 CAT 2022>
	at org.junit.Assert.fail(Assert.java:89)
	.....
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)

Reply:
Ok, then that's fine, I think I have run into this before too, one of them does not have the nano seconds

Reply:
Ok, I think I have run into this before, I wonder why one excludes the nano seconds

Reply:
I think it is a problem with junit with how it compares the date objects when they have nano seconds